### PR TITLE
Deprecate parts of read_dig_montage

### PR DESCRIPTION
**Pull request by massich**
works on #6461

### CODE REVIEW

**Comment by massich** [comment]
I wanted to rework `read_dig_montage` to not use `_read_dig_montage_XXXX` but the new code path through `read_XXX_dig` and remove the dead code. But its a useless effort. Is way more efficient to keep the old code, just warn and delete the whole thing after release. I'll just remove the dead code from fif.

**Comment by massich** [comment]
waiting for codecov to push the rest of the PR.

File: mne/channels/montage.py [hunk 1]
```diff
@@ -929,6 +974,9 @@ def read_dig_montage(hsp=None, hpi=None, elp=None,
         )
 
     elif bvct is not None:
+        warn('Using "read_dig_montage" with "bvct" not None'
+             ' is deprecated and will be removed in v0.20.'
+             ' Please use read_dig_captrack instead.', DeprecationWarning)
```
Comment (agramfort) [inline mne/channels/montage.py]: this is message is also in _read_dig_montage_deprecation_warning_helper ?
is this on purpose?

Comment (massich) [reply mne/channels/montage.py]: nop I just forgot to remove it.

**Comment by agramfort** [comment]
thx @massich

**Pull request merged**